CAIPs/caip-25.md

```diff
@@ -93,20 +130,28 @@ Example:
 ```
 
 The JSON-RPC method is labelled as `provider_authorization` and both the
-"requiredNamespaces" and "optionalNamespaces" arrays are populated with 
-`namespace` objects each named after the scope of authorization:
-1. EITHER an entire ChainAgnostic [namespace][] 
-2. OR a specific [CAIP-2][] in that namespace.
+"requiredScopes" and "optionalScopes" arrays are populated with 
+"scope objects" each named after the scope of authorization requested:
+1. EITHER an entire [CAIP-104][] [namespace][]
+2. OR a specific [CAIP-2][]-identified chain in a specific namespace.
 
-Each `namespace` object contains the following parameters:
+Each scope object contains the following parameters:
 - chains - array of [CAIP-2][]-compliant `chainId`'s. This parameter MAY be
   omitted if a single-chain scope is already declared in the index of the object.
 - methods - array of JSON-RPC methods expected to be used during the session
 - events - array of JSON-RPC message/events expected to be emitted during the
   session
 
-The `requiredNamespaces` array MUST contain 1 or more of these objects, if present; the `optionalNamespaces` array MUST contain 1 or more of them, if 
-present.
+The `requiredScopes` array MUST contain 1 or more of these objects, if present;
+the `optionalScopes` array MUST contain 1 or more of them, if present.
+
+There is one special-case scope object, named `wallet`, which refers to methods
+and events independent of any namespace but, crucially, no chains or CASA
```

Comment:
This essentially blocks wallet methods that are chain-related then no?

Reply:
no, `wallet_changeEthereumChain` would still be a valid method, but it would be in an `eip155` or `eip155:{chainId}` scope object!  the semantics/naming gets a little wonky but hey no one was thinking about non-ethereum wallets or offchain capabilities when the ethereum RPC methods were being named :D